1069140 - iSCSI devices for multipath have the same DeviceID

commit 4c62d144a516331cf7e206c9af7519472ffe35d3
Author: Jan Safranek <jsafrane@redhat.com>
Date:   Mon Feb 24 12:04:53 2014 +0100

    Use /dev/disk/by-path as DeviceID for iSCSI disks.
    
    When one LUN is imported several times (e.g. because of multipath), they
    get the same /dev/disk/by-id link and thus it cannot be used for DeviceID.

diff --git a/src/lmi/storage/util/storage.py b/src/lmi/storage/util/storage.py
index 8f1a3db..e6bbb4e 100644
--- a/src/lmi/storage/util/storage.py
+++ b/src/lmi/storage/util/storage.py
@@ -213,9 +213,17 @@ def get_persistent_name(device):
     :param device: (``StorageDevice``) The device.
     :returns: ``string``
     """
+    links = device.deviceLinks
+
     prefixes = ["/dev/disk/by-id/", "/dev/disk/by-partuuid/",
             "/dev/disk/by-path/", "/dev/disk/by-uuid/"]
-    links = device.deviceLinks
+
+    # use by-path for iscsi LUNs, they get the same by-id link
+    # if one is imported several times (e.g. for mulitpath)
+    if device.type == "iscsi":
+        prefixes = ["/dev/disk/by-path/", "/dev/disk/by-id/",
+                "/dev/disk/by-partuuid/", "/dev/disk/by-uuid/"]
+
     found = False
     # Find a symlink which matches the topmost prefix"
     for prefix in prefixes:
